drivers/interrupt_controller/intc_gicv3_priv.h

```diff
@@ -266,4 +266,16 @@ void its_rdist_invall(void);
 extern atomic_t nlpi_intid;
 #endif
 
+#if defined(CONFIG_ARMV8_A_NS) || defined(CONFIG_GIC_SINGLE_SECURITY_STATE)
+static ALWAYS_INLINE void arm_gic_write64(uint64_t val, mem_addr_t addr)
```

Comment:
uhm, I honestly don't like this. You are making a function called `arm_gic_write64` (so a pretty generic function to write a 64 bit register) conditional to something totally unrelated, like `CONFIG_ARMV8_A_NS` or `CONFIG_GIC_SINGLE_SECURITY_STATE`.

So either you change the name to something else or move the conditionals to `int_gicv3.c`.

Reply:
I just want to keep it specific for writing to IROUTER. What about if i change like this and move it to `intc_gicv3.c`
```
#if defined(CONFIG_ARMV8_A_NS) || defined(CONFIG_GIC_SINGLE_SECURITY_STATE)
static inline void arm_gic_write_irouter(uint64_t val, unsigned int intid)
{
	mem_addr_t addr = IROUTER(GET_DIST_BASE(intid), intid);

#ifdef CONFIG_ARM
	sys_write32((uint32_t)val, addr);
	sys_write32((uint32_t)(val >> 32U), addr + 4);
#else
	sys_write64(val, addr);
#endif
}
#endif
```

Reply:
yes, much better IMO

Reply:
Ok, updated